SMLHelper/Options/SliderModOption.cs

```diff
@@ -118,37 +137,56 @@ public class ModSliderOption : ModOption
         /// </summary>
         public float DefaultValue { get; }
 
-#if BELOWZERO
         /// <summary>
         /// The step value of the <see cref="ModSliderOption"/>.
         /// Defaults to 0.05f same as default code.
         /// </summary>
         public float Step { get; } = 0.05f;
-#endif
 
         /// <summary> Format for value field (<see cref="ModOptions.AddSliderOption(string, string, float, float, float, float?, string)"/>) </summary>
         public string ValueFormat { get; }
 
         private SliderValue sliderValue = null;
 
+#if SUBNAUTICA
+        private float previousValue;
```

Comment:
Btw, why is this needed? And only for SN?

Reply:
It is used here:
https://github.com/SubnauticaModding/SMLHelper/blob/88fc4806fd8b76cf5319a6fbc7d72df1965baefd/SMLHelper/Options/SliderModOption.cs#L162-L166

It tracks the previous value inside the custom SliderValue in SN (where we are implementing our own version of Slider step which hasn't reached SN yet) for triggering OnSliderChange, bc without it, we are trigging the change even when the *stepped to* value is the same bc the *original value* has changed ie.:
You have a slider that is set to step 5.
You move the slider from 0 to 5.
It triggers for 1, 2, 3, 4, 5, instead of just 5.
This behaviour is not what you would expect from a slider with a step of 5. You would expect it to trigger at 0, 5, 10, etc.
Tracking the previous value here mitigates that.

Reply:
Sorry, I remember it slightly wrong. It was actually triggering multiple times for the same, stepped, value, bc the underlying, non-stepped value has changed (0, 1, 2, 3 becomes 0, 0, 0, 5). So I am checking that the stepped-to value has actually changed before we trigger the OnSliderChange event.

Reply:
Ah, I see

Reply:
Once the stepped slider comes down the pipe from BZ to SN we can discard this.

We could also potentially just not offer a stepped slider in SN at all until the official one comes along - I'm fine with that too.

Reply:
Well, I added that step to example project because it's was easy way to show how `SliderValue` works, I don't think it's used for real. And I'm slightly worried that not calling `OnSliderChange` on the same value might break some code that expected that behaviour before.

Reply:
> And I'm slightly worried that not calling `OnSliderChange` on the same value might break some code that expected that behaviour before.

But they weren't using it before, because this is a new feature we are adding, so how can it? This is only used for stepped sliders - which don't exist in SN via SMLHelper until this update, unless you're using a custom slider, in which case you won't be using this code.

Reply:
I would also argue that a mod that is expecting to receive multiple repeats of the same value from a slider over and over and is *depending* on that... is an extremely strange (read: bad) mod for depending on such a thing.

Reply:
I mean, there was no check on value before, `OnSliderChange` was called always, even for the same value. On the other hand, I'm not sure that `UnityAction` even called for the same values, so it's probably fine.

Reply:
Oh, normal sliders *don't* send the same value over and over again. btw. I tested that. It was only an issue with this custom stepped implementation - so I made it work the same as normal - *not* sending the same value multiple times for no reason.